packages/flutter_tools/lib/src/vscode/vscode.dart

```diff
@@ -110,15 +115,26 @@ class VsCode {
   // macOS Extensions:
   //   $HOME/.vscode/extensions
   //   $HOME/.vscode-insiders/extensions
-  static List<VsCode> _installedMacOS(FileSystem fileSystem, Platform platform) {
+  static List<VsCode> _installedMacOS(FileSystem fileSystem, Platform platform, ProcessManager processManager) {
     final String? homeDirPath = FileSystemUtils(fileSystem: fileSystem, platform: platform).homeDirPath;
-    return _findInstalled(<_VsCodeInstallLocation>[
-      _VsCodeInstallLocation(
+    // Query Spotlight for unexpected installation locations.
+    final ProcessResult vsCodeSpotlightQueryResult = processManager.runSync(<String>[
```

Comment:
do we catch process exceptions in a higher scope? operating under the assumption this binary will either be removed or renamed some day.

Reply:
Good point.  Let me add that.

Reply:
Done.